src/net.cpp

```diff
@@ -113,6 +114,30 @@ std::map<CNetAddr, LocalServiceInfo> mapLocalHost GUARDED_BY(cs_mapLocalHost);
 static bool vfLimited[NET_MAX] GUARDED_BY(cs_mapLocalHost) = {};
 std::string strSubVersion;
 
+/**
+ * Check if making an automatic outbound connection to the given address is allowed.
+ * Could be restricted by the `-onlynet` configuration option.
+ * @return true if allowed
+ */
+static bool OutboundConnectionAllowedTo(const CNetAddr& addr)
+{
+    static const std::unordered_set<Network> onlynets = [](){
```

Comment:
This seems fragile. Can we guarantee `onlynets` is initialised only after the config files get loaded?

At the very least, I think it should be tested.

Reply:
Yes, I also strongly prefer initialization to be done explicitly in an explicit order. This code belongs in one of the `Init` functions, or called from there. We've historiically had many initialization bugs due to hidden gotchas like this.

Reply:
Dropped this altogether for a (hopefully) better and simpler approach.